.github/workflows/test-compilation.yml

```diff
@@ -0,0 +1,74 @@
+name: Test dd-trace-rb compilation
+
+on:
+  push:
+  workflow_call:
+
+jobs:
+  test-compilation:
+    runs-on: ubuntu-latest
+    name: "Ruby:${{ matrix.ruby_version }} expects ${{ matrix.expected_exit_code == 1 && '❌' || '✅' }} for ${{ matrix.ref }}"
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # Success cases (github.sha)
+          - { ruby_version: '3.4', ref: '${{ github.sha }}', expected_exit_code: 0 }
+          - { ruby_version: '3.3', ref: '${{ github.sha }}', expected_exit_code: 0 }
+          - { ruby_version: '3.2', ref: '${{ github.sha }}', expected_exit_code: 0 }
+          - { ruby_version: '3.1', ref: '${{ github.sha }}', expected_exit_code: 0 }
+          - { ruby_version: '3.0', ref: '${{ github.sha }}', expected_exit_code: 0 }
+          - { ruby_version: '2.7', ref: '${{ github.sha }}', expected_exit_code: 0 }
+          - { ruby_version: '2.6', ref: '${{ github.sha }}', expected_exit_code: 0 }
+          - { ruby_version: '2.5', ref: '${{ github.sha }}', expected_exit_code: 0 }
+
+          # Success cases (test-negative/missing-versions)
+          - { ruby_version: '3.2', ref: 'test-negative/missing-versions', expected_exit_code: 0 }
+          - { ruby_version: '3.1', ref: 'test-negative/missing-versions', expected_exit_code: 0 }
+          - { ruby_version: '3.0', ref: 'test-negative/missing-versions', expected_exit_code: 0 }
+          - { ruby_version: '2.7', ref: 'test-negative/missing-versions', expected_exit_code: 0 }
+          - { ruby_version: '2.6', ref: 'test-negative/missing-versions', expected_exit_code: 0 }
```

Comment:
I'm not sure it's worth testing with 2.6 to 3.2, since we know `datadog-ruby_core_source` isn't used on those.

Furthermore, it's not like we don't have test coverage for them -- the CI on dd-trace-rb side should pick up released updates for this gem and start running our tests with it on these rubies as well, so it'll flag issues on the really really really unlikely case we break something for them. 

So I think testing 2.6 to 3.2 is mostly a waste of CI cycles :eyes: